pkg/chunk/grpc/grpc.proto

```diff
@@ -0,0 +1,121 @@
+syntax = "proto3";
+
+package grpc;
+
+import "google/protobuf/empty.proto";
+
+service grpc_store {
+    //index-client
+    rpc BatchWrite(BatchWrites) returns (google.protobuf.Empty);
+    rpc QueryPages(IndexQuery) returns (stream ReadBatch);
+    rpc Delete(BatchDeletes) returns (google.protobuf.Empty);
+
+    //storage-client
+    rpc PutChunks(PutChunksRequest) returns (google.protobuf.Empty);
+    //In GetChunks rpc request send buf as nil
+    rpc GetChunks(GetChunksRequest) returns (stream GetChunksResponse);
+    rpc DeleteChunks(ChunkID) returns (google.protobuf.Empty);
+
+    //table-client
+    rpc ListTables(google.protobuf.Empty) returns (ListTablesResponse);
+    rpc CreateTable(TableDesc) returns (google.protobuf.Empty);
+    rpc DeleteTable(TableName) returns (google.protobuf.Empty);
+    rpc DescribeTable(TableName) returns (DescribeTableResponse);
+    rpc UpdateTable(UpdateTableRequest) returns (google.protobuf.Empty);
+
+    //used by storage-client & index-client
+    rpc Stop(google.protobuf.Empty) returns (google.protobuf.Empty);
+}
+
+message PutChunksRequest {
+    repeated Chunk chunks = 1;
+}
+
+message GetChunksRequest {
+    repeated Chunk chunks = 1;
+}
+
+message GetChunksResponse {
+    repeated Chunk chunks = 1;
+}
+
+message Chunk {
+    bytes encoded     = 1;
+    string key        = 2;
+    string tableName  = 3;
+}
+
+message ChunkID {
+    string chunkID = 1;
+}
+
+message TableName {
```

Comment:
A message `TableName` containing `tableName` looks a weird design to me. If you wanna keep it extensible, then I would split it into:

```
message DeleteTableRequest {
  string tableName = 1;
}

message DescribeTableRequest {
  string tableName = 1;
}
```